spring-cloud-stream/src/main/java/org/springframework/cloud/stream/provisioning/ProducerDestination.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.provisioning;
+
+import org.springframework.cloud.stream.binder.ProducerProperties;
+
+/**
+ * Represents a ProducerDestination that provides the information about the destination
+ * that is physically provisioned through
+ * {@link ProvisioningProvider#provisionProducerDestination(String, ProducerProperties)}
+ *
+ * @author Soby Chacko
+ */
+public interface ProducerDestination {
+
+	/**
+	 * Provides the destination name.
+	 *
+	 * @return destination name
+	 */
+	String getProducerDestinationName();
+
+	/**
+	 * Provides the destination name for a given partition
+	 *
+	 * @param partition the partition to find destination for
+	 * @return destination name for the given partition
+	 */
+	String getPartitionedProducerDestinationName(int partition);
```

Comment:
This contract smells for me somehow...

Maybe a JavaDocs would explain that there might be several destinations on the target broker because of partition logic?
Thanks